app/grandchallenge/reader_studies/forms.py

```diff
@@ -643,28 +651,55 @@ def __init__(self, *args, pk, interface, reader_study, user, **kwargs):
         qs = ComponentInterface.objects.exclude(
             slug__in=reader_study.values_for_interfaces.keys()
         )
+        widget_kwargs = {}
+        attrs = {}
+        if pk is None and selected_interface is not None:
+            widget = Select
+        else:
+            widget = autocomplete.ModelSelect2
+            attrs.update(
+                {
+                    "data-placeholder": "Search for an interface ...",
+                    "data-minimum-input-length": 3,
+                    "data-theme": settings.CRISPY_TEMPLATE_PACK,
+                    "data-html": True,
+                }
+            )
+            widget_kwargs[
+                "url"
+            ] = "components:component-interface-autocomplete"
+            widget_kwargs["forward"] = ["interface"]
+
         if pk is not None:
-            attrs = {
-                "hx-get": reverse_lazy(
-                    "reader-studies:display-set-interfaces-create",
-                    kwargs={"pk": pk, "slug": reader_study.slug},
-                ),
-                "hx-target": f"#ds-content-{pk}",
-            }
+            attrs.update(
+                {
+                    "hx-get": reverse_lazy(
+                        "reader-studies:display-set-interfaces-create",
+                        kwargs={"pk": pk, "slug": reader_study.slug},
+                    ),
+                    "hx-target": f"#ds-content-{pk}",
+                    "hx-trigger": "interfaceSelected",
```

Comment:
A shame that the simple `change` trigger doesn't work here, but nice work around!